public/main.js

```diff
@@ -0,0 +1,16 @@
+function fetchGET(url, cb) {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function() {
+        if (xhr.readySate === 4 && xhr.status === 200) {
+            cb(null, JSON.parse(xhr.responseText));
+        } else if (xhr.readyState === 4 && xhr.status !== 200) {
+            cb(xhr.status);
+        } else {
```

Comment:
Why do we need this else block??

Reply:
It's just my personal preference. I've used this function before in many projects so I know it works, if you have a suggestion to improve it feel free!